src/box/lua/config/descriptions.lua

```diff
@@ -1015,6 +1015,185 @@ I['failover.replicasets.*.learners.*'] = format_text([[
+    replication is configured per-space in tarantool, but the
+    supervised failover option is per-replicaset.
+
+    Technically a user can enable the `qsync` mode and mark only
+    some spaces synchronous. However, this setup doesn't give the
+    best of two worlds:
+
+    - Asynchronous transactions are assumed as possibly depending
+      from a result of previously started synchronous ones, so if
+      there are sync transactions in fly, all the async ones are
+      waiting for finishing of the sync ones. This way we
+      sometimes get sync-level latency for async transactions.
+    - Async transactions from the old leader, whose result is
+      visible to a client, are not necessarily present in another
+      availability zone. A network partitioning and repairing may
+      lead to conflicting journals and disconnecting of the old
+      leader from the replicaset until manual repairing activities
+      are performed. It means that the autofailover may hurt the
+      redundancy factor. And that happens frequently comparing to
+      the usual async setup due to the stricter conflict
+      detection strategy.
+
+    To sum up: in the mixed setup we pay twice (for async and
+    qsync both -- possible conflicts and increased latency), but
+    it rewards with a kind of 'sometimes you get better latency'
+    comparing to qsync and 'sometimes you have linearizable
+    writes' comparing to async. For most applications it seems
+    worse than fair qsync (with all the spaces marked as
+    synchronous).
+
+    In this description we assume either `async` with all
+    asynchronous spaces or `qsync` with all synchronous ones.
+
+    The `async` mode handles a replicaset with asynchronous
+    replication: if all the spaces are created without the
+    `is_sync` option. It is default.
+
+    The `qsync` mode handles a replicaset with quorum synchronous
+    replication: if the spaces have the `is_sync` option enabled.
+
+    Before describe how the modes work, let's introduce some
+    terms.
+
+    Imagine, a network partitioning happens. If the old leader is
+    not reachable from the active coordinator, it is *resigned*
+    (goes to read-only). The coordinator performs the
+    *autofailover*: it *appoints* the new leader (it goes to
+    read-write). The appoint may or may not claim *synchro queue*
+    ownership (to enable processing of synchronous transactions),
+    it may or may not require a *quorum* of replicas to agree
+    about the longest available journal.
+
+    Once the connectivity is repaired, the old leader attempts to
+    connect to the replicaset back. It succeeds or fails depending
+    on whether the journals are *diverged* and depending on the
+    *conflict detection* and the *conflict resolution* strategies.
+
+    The following description assumes such a network partitioning,
+    when the old leader lost connectivity with the active
+    coordinator, so the coordinator have to appoint a new leader.
```

Comment:
Side question: what if a leader is seen by the active coordinator, but it doesn't have the quorum to actually confirm any synchronous transactions? Will it keep being appointed as leader by the coordinator?